perception/lidar_transfusion/include/lidar_transfusion/transfusion_config.hpp

```diff
@@ -0,0 +1,158 @@
+// Copyright 2024 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef LIDAR_TRANSFUSION__TRANSFUSION_CONFIG_HPP_
+#define LIDAR_TRANSFUSION__TRANSFUSION_CONFIG_HPP_
+
+#include <cstdint>
+#include <vector>
+
+namespace lidar_transfusion
+{
+
+class TransfusionConfig
+{
+public:
+  TransfusionConfig(
+    const std::vector<int64_t> & voxels_num, const std::vector<double> & point_cloud_range,
+    const std::vector<double> & voxel_size, const float circle_nms_dist_threshold,
+    const std::vector<double> & yaw_norm_thresholds, const float score_threshold)
+  {
+    if (voxels_num.size() == 3) {
+      max_voxels_ = voxels_num[2];
+
+      voxels_num_[0] = voxels_num[0];
+      voxels_num_[1] = voxels_num[1];
+      voxels_num_[2] = voxels_num[2];
+
+      min_voxel_size_ = voxels_num[0];
+      opt_voxel_size_ = voxels_num[1];
+      max_voxel_size_ = voxels_num[2];
+
+      min_points_size_ = voxels_num[0];
+      opt_points_size_ = voxels_num[1];
+      max_points_size_ = voxels_num[2];
+
+      min_coors_size_ = voxels_num[0];
+      opt_coors_size_ = voxels_num[1];
+      max_coors_size_ = voxels_num[2];
+    }
+    if (point_cloud_range.size() == 6) {
+      min_x_range_ = static_cast<float>(point_cloud_range[0]);
+      min_y_range_ = static_cast<float>(point_cloud_range[1]);
+      min_z_range_ = static_cast<float>(point_cloud_range[2]);
+      max_x_range_ = static_cast<float>(point_cloud_range[3]);
+      max_y_range_ = static_cast<float>(point_cloud_range[4]);
+      max_z_range_ = static_cast<float>(point_cloud_range[5]);
+    }
+    if (voxel_size.size() == 3) {
+      voxel_x_size_ = static_cast<float>(voxel_size[0]);
+      voxel_y_size_ = static_cast<float>(voxel_size[1]);
+      voxel_z_size_ = static_cast<float>(voxel_size[2]);
+    }
+    if (score_threshold > 0.0) {
+      score_threshold_ = score_threshold;
+    }
+    if (circle_nms_dist_threshold > 0.0) {
+      circle_nms_dist_threshold_ = circle_nms_dist_threshold;
+    }
+    yaw_norm_thresholds_ =
+      std::vector<float>(yaw_norm_thresholds.begin(), yaw_norm_thresholds.end());
+    for (auto & yaw_norm_threshold : yaw_norm_thresholds_) {
+      yaw_norm_threshold =
+        (yaw_norm_threshold >= 0.0 && yaw_norm_threshold < 1.0) ? yaw_norm_threshold : 0.0;
+    }
+  }
+
+  ///// INPUT PARAMETERS /////
+  std::size_t cloud_capacity_ = 1000000;
```

Comment:
Please use uniform initialization
Constants should be declared  as such

Reply:
https://github.com/autowarefoundation/autoware.universe/pull/6890/commits/6ef95d3b6d013cc75c0184c537aefaba13543b44